### PR TITLE
OKTA-330325 Ensure primary authentication expiration is 5 minutes fro…

### LINKED ISSUE #41: AuthenticationResponse.ExpiresAt timezone parsing error?
##  Current behavior

Works fine in EST TZ (+5:00)

To reproduce the error, set computer UTC timezone (currently +1:00), try to 

`AuthenticationResponse = await authenticationClient.AuthenticateAsync(new AuthenticateOptions { Username = UserName, Password = Password } ...`

18:41:39.745 User XXX authenticated: MFA_REQUIRED: AuthenticationResponse.ExpiresAt == 9/15/2020 17:46:38 +01:00

55 minutes ago?

## Expected behavior

18:41:39.745 User XXX authenticated: MFA_REQUIRED: AuthenticationResponse.ExpiresAt == 9/15/2020 18:46:38 +01:00

in 5 minutes

## Minimal reproduction of the problem with instructions

Check the time here: https://github.com/OlegBoulanov/okta-auth-dotnet/blob/61487890f1afbb8fd3e10f32cc7a3c7e4405cd88/Okta.Auth.Sdk.UnitTests/AuthenticationClientShould.cs#L184

  | Name | Value | Type
-- | -- | -- | --
  | rawResponse | "\r\n            {\r\n                \"expiresAt\": \"2015-11-03T10:15:57.000Z\",... | string

  | Name | Value | Type
-- | -- | -- | --
▶ | ExpiresAt | {11/3/2015 10:15:57 -05:00} | System.DateTimeOffset?

## Environment

- OS: Windows 10

- .Net Framework: 4.61, 4.72

Related issue: https://github.com/okta/okta-sdk-dotnet/issues/347 ??

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
…m authentication

## Issue \#

https://oktainc.atlassian.net/browse/OKTA-330325
Fix #41 41

## Code

- [x] Unit test(s)
- [x] Implementation

## Current behavior

json date strings are being converted to DateTime objects, timezone information is lost during subsequent =>String=>DateTimeOffset conversion.

## Desired behavior

json date strings will be deserialized to DateTimeOffset objects, timezone information is preserved during String=>DateTimeOffset conversion.

### CODE REVIEW

**Review by bryanapellanes-okta — APPROVED** [review]
Glad this is fixable 👍

File: Okta.Sdk.Abstractions/DefaultSerializer.cs [hunk 1]
```diff
@@ -30,7 +30,8 @@ public DefaultSerializer()
                 NullValueHandling = NullValueHandling.Ignore,
                 DefaultValueHandling = DefaultValueHandling.Ignore,
                 ContractResolver = new DefaultContractResolver(),
-            };
+                DateParseHandling = DateParseHandling.DateTimeOffset,
+        };
```
Comment (laura-rodriguez) [inline Okta.Sdk.Abstractions/DefaultSerializer.cs:34]: Nit-pick: Fix indentation.

**Review by laura-rodriguez — APPROVED** [review]
One minor nit-pick to fix.

LGTM 👍

**Review by laura-rodriguez — APPROVED** [review]
Thanks!

**Pull request merged**